src/test/java/org/mule/extensions/vm/test/VMConsumeTestCase.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extensions.vm.test;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+import static org.mule.extensions.vm.api.VMError.EMPTY_QUEUE;
+import static org.mule.extensions.vm.api.VMError.QUEUE_NOT_FOUND;
+import static org.mule.runtime.api.metadata.DataType.JSON_STRING;
+import static org.mule.runtime.api.metadata.DataType.STRING;
+import static org.mule.tck.junit4.matcher.ErrorTypeMatcher.errorType;
+import org.mule.extensions.vm.internal.QueueListenerDescriptor;
+import org.mule.extensions.vm.internal.VMConnectorQueueManager;
+import org.mule.runtime.api.metadata.TypedValue;
+import org.mule.runtime.core.api.Event;
+import org.mule.runtime.core.api.exception.MessagingException;
+
+import java.io.Serializable;
+
+import org.junit.Test;
+
+public class VMConsumeTestCase extends VMTestCase {
```

Comment:
Not mandatory but we use Vm instead of VM (or Api instead of API) for names. Would be good to rename all of the VM usages.